projects/phone/src/lib/masks/phone/phone-mask-strict.ts

```diff
@@ -0,0 +1,67 @@
+import {MASKITO_DEFAULT_OPTIONS, MaskitoOptions} from '@maskito/core';
+import {
+    maskitoAddOnFocusPlugin,
+    maskitoCaretGuard,
+    maskitoPrefixPostprocessorGenerator,
+    maskitoRemoveOnBlurPlugin,
+} from '@maskito/kit';
+import {
+    AsYouType,
+    CountryCode,
+    getCountryCallingCode,
+    MetadataJson,
+} from 'libphonenumber-js/core';
+
+import {
+    cutInitCountryCodePreprocessor,
+    phoneLengthPostprocessorGenerator,
+    validatePhonePreprocessorGenerator,
+} from './processors';
+import {generatePhoneMask, getPhoneTemplate} from './utils';
+
+export function maskitoPhoneStrictOptionsGenerator({
+    countryIsoCode,
+    metadata,
+}: {
+    countryIsoCode: CountryCode;
+    metadata: MetadataJson;
+}): Required<MaskitoOptions> {
+    const code = getCountryCallingCode(countryIsoCode, metadata);
+    const formatter = new AsYouType(countryIsoCode, metadata);
+    const prefix = `+${code} `;
+
+    let template = '';
+    let prevPhoneLength = 0;
+
+    return {
+        ...MASKITO_DEFAULT_OPTIONS,
+        mask: ({value}) => {
+            const newTemplate = getPhoneTemplate(formatter, value);
+            const phoneLength = value.replace(/\D/g, '').length;
+
+            template =
```

Comment:
Why do these change for a strict mask? I wish there was a way to reuse this piece of code :)